src/TLuaInterpreterMapper.cpp

```diff
@@ -3479,12 +3484,32 @@ int TLuaInterpreter::setRoomArea(lua_State* L)
         return warnArgumentValue(L, __func__, "no map present or loaded");
     }
 
-    const int id = getVerifiedInt(L, __func__, 1, "roomID");
-    if (!host.mpMap->mpRoomDB->getRoomIDList().contains(id)) {
-        return warnArgumentValue(L, __func__, csmInvalidRoomID.arg(id));
+    QVector<int> roomIds;
+    if (lua_isnumber(L, 1)) {
+        const int id = getVerifiedInt(L, __func__, 1, "roomID");
+        if (!host.mpMap->mpRoomDB->getRoomIDList().contains(id)) {
+            return warnArgumentValue(L, __func__, csmInvalidRoomID.arg(id));
+        }
+        roomIds.append(id);
+    } else if (lua_istable(L, 1)) {
+        lua_pushnil(L);
+        while (lua_next(L, 1) != 0) {
+            const int id = getVerifiedInt(L, __func__, -1, "roomID");
+            if (!host.mpMap->mpRoomDB->getRoomIDList().contains(id)) {
+                return warnArgumentValue(L, __func__, csmInvalidRoomID.arg(id));
```

Comment:
Would it not be reasonable to identify the index (starting at 1 obviously - it *is* a *Lua* table) of the invalid room id that prompted the error - or do you consider it sufficient to report only the bogus value?

Reply:
I think just thr wrong value is enough